cmd/gardener-node-agent/app/app.go

```diff
@@ -227,7 +261,41 @@ func run(ctx context.Context, cancel context.CancelFunc, log logr.Logger, cfg *c
 	return mgr.Start(ctx)
 }
 
-func getRESTConfig(log logr.Logger, cfg *config.NodeAgentConfiguration) (*rest.Config, bool, error) {
+func getRESTConfig(log logr.Logger, fs afero.Afero, cfg *config.NodeAgentConfiguration) (*rest.Config, bool, error) {
+	if kubeconfig, err := fs.ReadFile(nodeagentv1alpha1.KubeconfigFilePath); err == nil {
+		log.Info("Kubeconfig file exists, using it")
+		restConfig, err := kubernetes.RESTConfigFromKubeconfig(kubeconfig)
+		if err != nil {
+			return nil, false, err
+		}
+		return restConfig, false, nil
+	} else if !errors.Is(err, afero.ErrFileNotFound) {
+		return nil, false, err
+	}
+
+	restConfig := &rest.Config{
+		Burst: int(cfg.ClientConnection.Burst),
+		QPS:   cfg.ClientConnection.QPS,
+		ContentConfig: rest.ContentConfig{
+			AcceptContentTypes: cfg.ClientConnection.AcceptContentTypes,
+			ContentType:        cfg.ClientConnection.ContentType,
+		},
+		Host:            cfg.APIServer.Server,
+		TLSClientConfig: rest.TLSClientConfig{CAData: cfg.APIServer.CABundle},
+		BearerTokenFile: nodeagentv1alpha1.BootstrapTokenFilePath,
+	}
+
+	if _, err := fs.Stat(nodeagentv1alpha1.BootstrapTokenFilePath); err != nil && !errors.Is(err, afero.ErrFileNotFound) {
+		return nil, false, fmt.Errorf("failed checking whether bootstrap token file %q exists: %w", nodeagentv1alpha1.BootstrapTokenFilePath, err)
+	} else if err == nil {
+		log.Info("Kubeconfig file does not exist, but bootstrap token file does - using it", "path", nodeagentv1alpha1.BootstrapTokenFilePath)
+		return restConfig, true, nil
+	}
```

Comment:
Can this block be written as
```suggestion
	if ok, err := fs.Exists(nodeagentv1alpha1.BootstrapTokenFilePath); err != nil {
		return nil, false, fmt.Errorf("failed checking whether bootstrap token file %q exists: %w", nodeagentv1alpha1.BootstrapTokenFilePath, err)
	} else if ok {
		log.Info("Kubeconfig file does not exist, but bootstrap token file does - using it", "path", nodeagentv1alpha1.BootstrapTokenFilePath)
		return restConfig, true, nil
	}
```
?